launcher/ui/GuiUtil.cpp

```diff
@@ -51,11 +51,35 @@
 #include <settings/SettingsObject.h>
 #include "Application.h"
 
+constexpr int MaxMclogsLines = 25000;
+constexpr int InitialMclogsLines = 10000;
+constexpr int FinalMclogsLines = 14900;
+
+QString truncateLogForMclogs(const QString& logContent)
+{
+    QStringList lines = logContent.split("\n");
+    if (lines.size() > MaxMclogsLines) {
+        QString truncatedLog = lines.mid(0, InitialMclogsLines).join("\n");
+        truncatedLog +=
+            "\n\n\n\n\n\n\n\n\n\n"
+            "------------------------------------------------------------\n"
+            "--------------------- Log truncated by ---------------------\n"
+            "---------------------- Prism Launcher ----------------------\n"
```

Comment:
Do we need these two lines... like a said we have a constant to be used for the app name (I missed this :P) but this would require an algorithm to pad it which might be surprisingly difficult? Is this to make it stand out more?

Reply:
i thought smth like this would be a good idea to make it clear the log was cut off and figured i might as well mention who's responsible for it, i'd say it's fine to just make it say "log truncated" without mentioning prism if you'd prefer this

Reply:
I guess it's no big deal, lmk if you think it does add something

Reply:
Sorry my GitHub kept not updating with your comments

Reply:
Fair enough, I don't think we have many forks other than you-know-what at the moment, they'd just better be checking what they merge in lol

Reply:
oh uhh sorry my github wasn't updating either 😓